pulsar-zookeeper-utils/src/main/java/org/apache/pulsar/zookeeper/ZkBookieRackAffinityMapping.java

```diff
@@ -162,10 +180,19 @@ private String getRack(String bookieAddress) {
             throw new RuntimeException(e);
         }
 
+        BookieInfo bi = bookieInfoMap.get(bookieAddress);
+        if (bi == null) {
+            Optional<BookieInfo> biOpt = racksWithHost.getBookie(bookieAddress);
+            if (biOpt.isPresent()) {
+                bi = biOpt.get();
+            } else {
+                updateRacksWithHost(racksWithHost);
```

Comment:
Do we need to update it here? seems this is a  get operation, If it is not present, there is no need to update?